src/js/components/TextInput.js

```diff
@@ -41,18 +41,7 @@ export default class TextInput extends Component {
     const { suggestions } = this.props;
     const { announceChange, dropActive, focused } = this.state;
     const { intl } = this.context;
-    // Set up keyboard listeners appropriate to the current state.
-    const activeKeyboardHandlers = {
-      esc: this._onRemoveDrop,
-      tab: this._onRemoveDrop,
-      up: this._onPreviousSuggestion,
-      down: this._onNextSuggestion,
-      enter: this._onEnter
-    };
-    const focusedKeyboardHandlers = {
-      down: this._onAddDrop
-    };
-
+    const {activeKeyboardHandlers, focusedKeyboardHandlers} = this;
```

Comment:
Great contribution. I think we can simplify it a little bit:

```
export default class TextInput extends Component {
  activeKeyboardHandlers =  {
      esc: this._onRemoveDrop,
      tab: this._onRemoveDrop,
      up: this._onPreviousSuggestion,
      down: this._onNextSuggestion,
      enter: this._onEnter
  };

  focusedKeyboardHandlers =  {
      down: this._onAddDrop
  };

  componentDidUpdate (prevProps, prevState) {
      // use `this.activeKeyboardHandlers` and `this.focusedKeyboardHandlers`
  } 

  componenWillUnmount() {
     // use `this.activeKeyboardHandlers` and `this.focusedKeyboardHandlers`
  }
}
```

Reply:
I think syntactically you have to include the get directive on them in ES6. I'm not sure you can directly declare class variables (although I've seen it suggested in ES7). Do we have examples of this being done in Grommet today?
eslint gives an error unless I put it below the lifecycle methods. You can disable this in eslint but I don't see it disabled in the eslint file.

Reply:
Ok, it should be up to date now